### PR TITLE
Add docs on securing webhooks with service accounts

### PR DESCRIPTION
**Pull request by bunchesofdonald**
Adds documentation on securing webhooks with service accounts in Prefect Cloud.

### CODE REVIEW

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 1]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
 there is any error interpreting the request. For more visibility when your webhooks
 fail, see the [Troubleshooting](#troubleshooting) section below.
 
+
+## Webhook Authentication
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:101]:
```suggestion
## Webhook authentication
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 2]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+
+To enhance security, within a Prefect Cloud Pro or Enterprise tier account, you
+can assign a Service Account to a Webhook. This feature provides an additional
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:104]:
```suggestion
can assign a service account to a webhook. This feature provides an additional
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 3]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+layer of security and is highly recommended, especially for enterprise
+environments. 
+
+### API Key Authentication
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:108]:
```suggestion
### API key authentication
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 4]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+
+Webhooks can be associated with a Service Account, and the webhook endpoint
+will be authenticated using the Service Account's API Key.
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:111]:
```suggestion
will be authenticated using the service account's API key. 
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 5]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+can assign a Service Account to a Webhook. This feature provides an additional
+layer of security and is highly recommended, especially for enterprise
+environments. 
+
+### API Key Authentication
+
+Webhooks can be associated with a Service Account, and the webhook endpoint
+will be authenticated using the Service Account's API Key. 
+
+To use API Key authentication:
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:113]:
```suggestion
To use API key authentication:
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 6]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+
+1. Create a Service Account or use an existing one.
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:115]:
```suggestion
1. Create a service account or use an existing one.
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 7]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+2. Associate the Service Account with your webhook from the UI.
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:116]:
```suggestion
2. Associate the service account with your webhook from the UI.
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 8]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+3. Include the API Key in the `Authorization` header of your webhook requests:
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:117]:
```suggestion
3. Include the API key in the `Authorization` header of your webhook requests:
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 9]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+
+```
+Authorization: Bearer <your-api-key>
+```
+
+### Account-Level Security Enforcement
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:123]:
```suggestion
### Account-level security enforcement
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 10]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+
+An account-level setting is available to enforce authentication on all webhooks. When enabled:
+
+- All existing webhooks without an associated Service Account will be automatically disabled.
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:127]:
```suggestion
- All existing webhooks without an associated service account will be automatically disabled.
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 11]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+- New webhooks must be created with an associated Service Account.
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:128]:
```suggestion
- New webhooks must be created with an associated service account.
```

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 12]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+- Existing webhooks must be updated to include a Service Account before they can be re-enabled.
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:129]:
```suggestion
- Existing webhooks must be updated to include a service account before they can be re-enabled.
```

**Review by discdiver — COMMENTED** [review]
Thank you! Adjusted capitalization to match other docs.

**Review by dylanbhughes — COMMENTED** [review]
Content LGTM 👍🏻

**Comment by bunchesofdonald** [comment]
> Thank you! Adjusted capitalization to match other docs.

Thanks @discdiver! Those are all committed.

File: docs/3.0/automate/events/webhook-triggers.mdx [hunk 13]
```diff
@@ -97,6 +97,38 @@ a `204 No Content` response when they are successful or a `400 Bad Request` erro
...
+## Webhook authentication
+
+To enhance security, within a Prefect Cloud Pro or Enterprise tier account, you
+can assign a service account to a webhook. This feature provides an additional
+layer of security and is highly recommended, especially for enterprise
+environments. 
+
+### API key authentication
+
+Webhooks can be associated with a Service Account, and the webhook endpoint
```
Comment (discdiver) [inline docs/3.0/automate/events/webhook-triggers.mdx:110]:
```suggestion
Webhooks can be associated with a service account, and the webhook endpoint
```

**Review by discdiver — APPROVED** [review]
One more I missed below.

**Review by desertaxle — APPROVED** [review]

**Pull request merged**